src/boot/errorHandling.js

```diff
@@ -67,68 +69,185 @@ const html = `
         </div>
     </div>
 `;
-const successfulTransactionNotification = function(link) {
-    Notify.create({
-        timeout: 5000,
+
+const notifyMessage = function(type, icon, title, message, payload) {
+
+    // action buttons
+    const actions = [];
+    const dismiss_btn = {
+        label: this.$t('notification.dismiss_label'),
+        class: 'c-notify__action-btn',
+    };
+    const link_btn = {
+        label: this.$t('notification.success_see_trx_label'),
+        color: 'positive',
+        iconRight: 'launch',
+        class: 'c-notify__action-btn',
+        handler: () => {
+            window.open(payload, '_blank');
+        },
+    };
+    const details_btn = {
+        label: this.$t('notification.error_see_details_label'),
+        class: 'c-notify__action-btn ',
+        handler: () => {
+            let content = '';
+            // We try to parse the payload to obtain a content string
+            try {
+                // First we try to parse the payload as JSON
+                content = JSON.stringify(payload, null, 2);
+            } catch (e) {
+                // If it fails, we discard the error and try to executes toString() from the payload
+                try {
+                    content = payload.toString();
+                } catch (e) {
+                    // If it fails, likely the payload is null (or some unknown object with no toString function),
+                    // so we set the content to a forced string to have something to show
+                    content = payload + ' ';
+                }
+            }
```

Comment:
I added some comments to explain what is going on and why.
This is again in response to @karynemayer's [feedback](https://github.com/telosnetwork/telos-wallet/pull/373#discussion_r1218349171) on the original PR.